Add subsection on edge prior results (task 1/3) (#24)
greenelab/xswap-manuscript: content/02.body.md

## Results

### Edge prior

In the first prediction task, we computed three features---the XSwap edge prior, an approximation to the edge prior, and the (scaled) product of source and target node degree---on networks from Hetionet.

> We used the features in attempting to reconstruct the 20 networks themselves.

Reviewer: ```suggestion
We then evaluated the extent to which these features could reconstruct the 20 networks.
```

> The XSwap-derived edge prior is able to reconstruct many of the networks with a high level of performance, as measured by the AUROC.

Reviewer: ```suggestion
The XSwap-derived edge prior reconstructed many of the networks with a high level of performance, as measured by the AUROC.
```

Of the 20 individual networks we extracted from Hetionet, 17 had an edge prior self-reconstruction AUROC >= 0.95, with the highest reconstruction AUROC at 0.9971 (Compound–downregulates–Gene edge type).
Meanwhile, the lowest self-reconstruction performance (AUROC = 0.7697, Disease–localizes–Anatomy edge type) occurred in the network having the fewest node pairs.

![AUROC of network reconstruction by prediction task.
  The edge prior shows strong performance for network reconstruction when computed on the original (Task 1) and sampled (Task 2) networks.
  The performance reduction from computing features on sampled networks is real but far smaller compared to a new degree distribution.
  ](https://github.com/zietzm/xswap-analysis/raw/862d9504ec8546ea998285a708ab53a6c4f2fe2f/img/auroc_dists.png){#fig:discrimination width="50%"}

> The three features used in the comparison are highly rank correlated (median > 0.99999 across metaedges).

Reviewer: ```suggestion
The three features that we compared were highly rank correlated (median > 0.99999 across metaedges).
```

> Because AUROC takes only rank into account, the three features have very similar AUROC reconstruction performance values for the first, second, and third prediction tasks (max difference < 0.027).

Reviewer: ```suggestion
The three features also had very similar AUROC reconstruction performance values for the first, second, and third prediction tasks (max difference < 0.027) because AUROC is rank-based.
```

> Nonetheless, we found that the edge prior exhibited marginally superior discriminatory performance in 12 of 20 networks.

Reviewer: ```suggestion
The edge prior was slightly better than the approximations in 12 of 20 networks.
```

> While the discriminatory abilities are similar, the features are very different in their levels of calibration---the ability of the model to correctly estimate class membership probabilities.

Reviewer: ```suggestion
However, while the AUROC results were similar, the features were very different in their levels of calibration---the ability of the model to correctly estimate class membership probabilities.
```

> We find that the edge prior is very well calibration for all networks in the first and second tasks, and it shows the best calibration of the three features for each of the prediction tasks (Figure {@fig:calibration}A).

Reviewer: ```suggestion
We found that the edge prior was very well calibrated for all networks in the first and second tasks, and it provided the best calibration of the three features for each of the prediction tasks (Figure {@fig:calibration}A).
```

> As the edge prior is not based on the networks' true edges, these results indicate that degree sequence alone is highly informative and can allow the reconstruction of edges, and they validate the use of the edge prior for this purpose.

Reviewer: ```suggestion
As the edge prior was not based on the networks' true edges, these results indicated that degree sequence alone was highly informative and that permutation was the only approach that provided a well-calibrated model.
```

> ![**A.** The edge prior shows superior calibration to the other two features.

Reviewer: ```suggestion
![**A.** The permutation-based edge prior's calibration was superior to the two strategies based on estimates.
```

>   Calibration curves for full network reconstruction.

Reviewer: ```suggestion
  Calibration curves for full network reconstruction of 20 networks from Hetionet.
```

>   20 networks shown are from Hetionet.

Reviewer: ```suggestion
```

>   **B.** Calibration curves for sampled network reconstruction. The edge prior shows superior calibration, though the levels of calibration decrease for all three features.

Reviewer: ```suggestion
  **B.** Calibration curves for sampled network reconstruction. The edge prior shows superior calibration in the 20 Hetionet networks.
```

>   20 networks shown are from Hetionet.

Reviewer: ```suggestion
```

  **C.** Individual Hetionet edge type calibration metrics.

>   Lower scores indicate better calibration.

Reviewer: I don't understand what is shown on this plot. Is the Y axis the error in the calibration?

Reviewer: Instead of using "calibration" as a noun, could you try writing this with saying one is calibrated better or worse than the others? It might be a little bit clearer to read.

The edge prior was slightly better than the approximations in 12 of 20 networks.
However, while the AUROC results were similar, the features were very different in their levels of calibration---the ability of the model to correctly estimate class membership probabilities.
We found that the edge prior was very well calibrated for all networks in the first and second tasks, and it provided the best calibration of the three features for each of the prediction tasks (Figure {@fig:calibration}A).
As the edge prior was not based on the networks' true edges, these results indicated that degree sequence alone was highly informative and that permutation was the only approach that provided a well-calibrated model.

![**A.** Calibration curves for full network reconstruction of 20 networks from Hetionet.
  The permutation-based edge prior's calibration was superior to the other two strategies based on degree.
  **B.** Calibration curves for sampled network reconstruction.
  The edge prior shows superior calibration in the 20 Hetionet networks.

>   **C.** Individual Hetionet edge type calibration metrics.

Reviewer: ```suggestion
  **C.** Individual Hetionet edge type calibration estimated by the two-component decomposition of the Brier score, in which lower scores indicate better calibration.
```

>   Lower scores indicate better calibration.

Reviewer: ```suggestion
```

>   Edge prior shows excellent calibration in unsampled and sampled networks, but inferior calibration to scaled degree for prediction across different degree distributions.

Reviewer: ```suggestion
  The edge prior has excellent calibration in unsampled and sampled networks, and each considered method is sensitive to shifts in the degree distribution.
```

>   Calibration metric is from the two-component decomposition of the Brier score, and a lower score indicates superior calibration.](https://github.com/zietzm/xswap-analysis/raw/fb5e093ab5baedd129629ee8e41526c286cbbe45/img/fig4.calibration.png){#fig:calibration width="100%"}

Reviewer: ```suggestion
](https://github.com/zietzm/xswap-analysis/raw/fb5e093ab5baedd129629ee8e41526c286cbbe45/img/fig4.calibration.png){#fig:calibration width="100%"}
```
I think you may also have to move this up a line.

Another thing that may be helpful - what about fixing the Y-axis range across all 3 of the small multiples instead of just 2? The errors with the degree shift seem like they get pretty large when compared to the unsampled and sampled networks.
